app/src/org/commcare/activities/FormRecordListActivity.java

```diff
@@ -436,16 +443,38 @@ public void run() {
         }
     }
 
-    private void createFormRecordScanResultDialog(Pair<Boolean, String> result) {
+    private void createFormRecordScanResultDialog(Pair<Boolean, String> result, FormRecord record) {
         String title;
         if (result.first) {
             title = Localization.get("app.workflow.forms.scan.title.valid");
         } else {
             title = Localization.get("app.workflow.forms.scan.title.invalid");
         }
         int resId = result.first ? R.drawable.checkmark : R.drawable.redx;
-        showAlertDialog(StandardAlertDialog.getBasicAlertDialogWithIcon(this, title,
-                result.second, resId, null));
+
+        StandardAlertDialog dialog = StandardAlertDialog.getBasicAlertDialogWithIcon(this, title,
+                result.second, resId, null);
+
+        dialog.setNegativeButton(Localization.get("app.workflow.forms.quarantine"), new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                quarantineRecord(record);
+                dismissAlertDialog();
+            }
+        });
+
+        showAlertDialog(dialog);
+    }
+
+    private void quarantineRecord(FormRecord record) {
+        this.formRecordProcessor.updateRecordStatus(record, FormRecord.STATUS_QUARANTINED);
```

Comment:
By "in the same place", I assume you mean under the same log tag? Quarantines due to a local record issue are [currently logged under the tag `"error-storage"`](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/ProcessAndSendTask.java#L317) which I don't think is a semantically correct place to log this new type of quarantine. I have just updated the log message for the old log though, to specify the cause as being "due to local problem with record"